pypeit/core/wavecal/autoid.py

```diff
@@ -770,50 +818,94 @@ def cross_match(self, good_fit):
             lindex = np.array([], dtype=np.int)
             dindex = np.array([], dtype=np.int)
             wcen = np.zeros(good_slits.size)
+            disp = np.zeros(good_slits.size)
+            shift_vec = np.zeros(good_slits.size)
+            stretch_vec = np.zeros(good_slits.size)
+            ccorr_vec = np.zeros(good_slits.size)
             for cntr, gs in enumerate(good_slits):
+                msgs.info('Cross-correlating bad slit # {:d}'.format(bs + 1) + ' with good slit # {:d}'.format(gs + 1))
                 # Match the peaks between the two spectra.
                 # spec_gs_adj is the stretched spectrum
-                stretch, shift = wvutils.match_peaks(self._spec[:, bs], self._spec[:, gs])
-                if stretch is None:
+                success, shift_vec[cntr], stretch_vec[cntr], ccorr_vec[cntr], _, _ =  \
+                    wvutils.xcorr_shift_stretch(self._spec[:, bs],self._spec[:, gs], debug = self._debug)
+                if not success:
                     continue
-                # Estimate wcen for this slit
-                wcen[cntr] = self._all_patt_dict[str(gs)]['bwv'] - shift*disp
-                # For each peak in the gs spectrum, identify the corresponding peaks in the bs spectrum
-                strfact = (self._npix + stretch - 1)/(self._npix - 1)
+                # JFH Put in a cut on the cross-correlation value here in this logic
```

Comment:
I think that would be a good idea but can wait for a future PR.
You might capture it as an Issue or at least a TODO here.

Reply:
Listed it as a ToDo.